fs-index/src/index.rs

```diff
@@ -491,10 +502,15 @@ impl<Id: ResourceId> ResourceIndex<Id> {
     pub fn update_one<P: AsRef<Path>>(
```

Comment:
Also, I'm not sure that we still require that added resource isn't in the index yet. What happens if it's already tracked? Do we silently consider it modification and update its timestamp? We could add assertion just to prevent mis-use.

Reply:
> It seems, that modification of content by same path doesn't impose any invariants to check before the call.

Exactly. If the developer calls `update_one()` expecting to modify a file that isn’t actually in the index, it will simply treat it as a new addition.


> what happens if this sequence is not followed

At any point in time, if `update_one()` is called with a path that doesn’t exist either in the file system or in the index, it will panic. I noticed this while implementing the file system watcher without the debouncer; it picks up a lot of noisy events, and calling `update_one()` on them causes these panics

>  What happens if it's already tracked?

If the file is already tracked and `update_one()` is called, it’s assumed to be a modification. The file will simply be rescanned and updated in the index without changing its timestamp, since there’s no actual update to the file. 
In other words, calling `update_one()` multiple times on a resource that exists in both the file system and the index has no real impact or downside.